language/move-vm/runtime/src/interpreter.rs

```diff
@@ -732,19 +844,25 @@ const OPERAND_STACK_SIZE_LIMIT: usize = 1024;
 const CALL_STACK_SIZE_LIMIT: usize = 1024;
 
 /// The operand stack.
-struct Stack(Vec<Value>);
+struct Stack {
+    value: Vec<Value>,
+    types: Vec<ParanoidType>,
```

Comment:
Stacks now have a vector of types along side the value.

Reply:
Nice, this seems less disruptive than having a stack of (Value, Type) pairs